jwst/associations/lib/rules_level3.py

```diff
@@ -112,14 +114,194 @@ def __init__(self, *args, **kwargs):
             ),
             Constraint_Optical_Path(),
             Constraint_Target(),
+            DMSAttrConstraint(
+                name='exp_type',
+                sources=['exp_type'],
+                value=(
+                    'mir_lrs-fixedslit'
+                    '|mir_lrs_slitless'
+                ),
+                force_unique=False
+            )
+        ])
+
+        # Check and continue initialization.
+        super(Asn_SpectralTarget, self).__init__(*args, **kwargs)
+
+
+@RegistryMarker.rule
+class Asn_SpectralSource(AsnMixin_Spectrum):
+    """Slit-like, multi-object spectrographic modes"""
+
+    def __init__(self, *args, **kwargs):
+
+        # Setup for checking.
+        self.constraints = Constraint([
+            Constraint(
+                [Constraint_TSO()],
+                reduce=Constraint.notany
+            ),
+            Constraint_Optical_Path(),
+            Constraint_Target(),
+            Constraint(
+                [
+                    DMSAttrConstraint(
+                        name='exp_type',
+                        sources=['exp_type'],
+                        value=(
+                            'nrc_grism'
```

Comment:
``nrc_grism`` has been relabeled to ``nrc_wfss``, which is already in the list, so ``nrc_grism`` could be removed. Also, ``nrc_tsgrism`` likely belongs in the other list, because it's a single target slitless mode.